hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -104,13 +104,14 @@ public class HoodieMetadataPayload implements HoodieRecordPayload<HoodieMetadata
   private static final Logger LOG = LoggerFactory.getLogger(HoodieMetadataPayload.class);
   /**
    * Type of the record. This can be an enum in the schema but Avro1.8
-   * has a bug - https://issues.apache.org/jira/browse/AVRO-1810
+   * has a bug - <a href="https://issues.apache.org/jira/browse/AVRO-1810">...</a>
    */
-  protected static final int METADATA_TYPE_PARTITION_LIST = 1;
-  protected static final int METADATA_TYPE_FILE_LIST = 2;
-  protected static final int METADATA_TYPE_COLUMN_STATS = 3;
-  protected static final int METADATA_TYPE_BLOOM_FILTER = 4;
+  private static final int METADATA_TYPE_PARTITION_LIST = 1;
+  private static final int METADATA_TYPE_FILE_LIST = 2;
+  private static final int METADATA_TYPE_COLUMN_STATS = 3;
+  private static final int METADATA_TYPE_BLOOM_FILTER = 4;
   private static final int METADATA_TYPE_RECORD_INDEX = 5;
+  private static final int METADATA_TYPE_PARTITION_STATS = 6;
```

Comment:
Maybe we should add enum and guarantee the ordering, and automatically assign the type ID?

Reply:
Hmm.. yes we could. Just to be clear, you're suggesting no schema change right? As in we will still keep the int value in the records, and not enum.

Reply:
Right, no schema change; just code restructuring maintaining the same ID, i.e., add `getId` in `MetadataPartitionType`.  You can refactor this part in a separate PR.